README.md

```diff
@@ -1,2 +1,243 @@
+
+### Testnet versions
+
+- [x] **Local Testnet** is a developer local setup, mono-node, assembling SIP 001, SIP 002, SIP 004 and SIP 005. With this version, developers can run Stack 2.0 on their development machines, but also write, execute and tests smart contracts. See the instructions below for more details.
+
+- [ ] **Open Testnet** is the incoming version of our testnet, that we're intending to ship in Q1 2020. This testnet will ship with SIP 003 and be an open membership network, where participants will be able to observe and experience the consensus running at scale.
+
+- [ ] **Mainet** is the fully functional version, that we're intending to ship in Q2 2020.
+
+## Getting started
+
+### Download and build blockstack-core
+
+The first step is to ensure that you have Rust and the support software installed.
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+From there, you can clone this repository:
+
+```bash
+git clone https://github.com/blockstack/blockstack-core.git
+
+cd blockstack-core
+```
+
+Then build the project:ca
+
+```bash
+cargo build
+```
+
+And run the tests:
+
+```bash
+cargo test testnet  -- --test-threads=1
+```
+
+### Encode and sign transactions
+
+Let's start by generating a keypair, that will be used for signing the upcoming transactions:
+
+```bash
+cargo run --bin blockstack-cli generate-sk --testnet
+
+# Output
+# {
+#  secretKey: "b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001",
+#  publicKey: "02781d2d3a545afdb7f6013a8241b9e400475397516a0d0f76863c6742210539b5",
+#  stacksAddress: "ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH"
+# }
+```
+
+We are considering the following simple contract `kv-store`, that you can save to `./kv-store.clar`:
+
+```scheme
+(define-map store ((key (buff 32))) ((value (buff 32))))
+
+(define-public (get-value (key (buff 32)))
+    (match (map-get? store ((key key)))
+        entry (ok (get value entry))
+        (err 0)))
+
+(define-public (set-value (key (buff 32)) (value (buff 32)))
+    (begin
+        (map-set store ((key key)) ((value value)))
+        (ok 'true)))
+```
+
+We want to publish this contract on chain, and issuing some transactions interacting with it, setting some keys and getting some values, so we can observe read and writes.
+
+Our first step is to generate and sign, using your private key, the transaction that will publish the contract `kv-store`.
+To do that, we will use the subcommand:
+
+```bash
+cargo run --bin blockstack-cli publish --help
+```
+
+With the following arguments:
+
+```bash
+cargo run --bin blockstack-cli publish b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001 0 0 kv-store ./kv-store.clar --testnet
+```
+
+This command will output the **binary format** of the transaction. In our case, we want to pipe this output and dump it to a file that will be used later in this step by step.
```

Comment:
"...will be used later in this tutorial."

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1